### PR TITLE
feat: add casbin extension

### PR DESCRIPTION
**Pull request by L2ncE**
#### What type of PR is this?

feat

#### What this PR does / why we need it (en: English/zh: Chinese):

en: add casbin extension
zh: 新增 casbin 拓展

#### Which issue(s) this PR fixes:

https://github.com/cloudwego/hertz-examples/pull/75

### CODE REVIEW

File: casbin.go [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+			return
+		}
+		// Enforce Casbin policies.
+		if options.Logic == AND {
+			// Must pass all tests.
+			for _, permission := range permissions {
+				vals := append([]string{sub}, options.PermissionParser(permission)...)
+				if vals[0] == "" || vals[1] == "" {
+					// Can not handle any illegal permission strings.
+					c.AbortWithStatus(http.StatusInternalServerError)
```
Comment (li-jin-gou) [inline casbin.go]: use `consts` inestead of `net/http`

Comment (li-jin-gou) [reply casbin.go]: https://github.com/cloudwego/hertz/tree/develop/pkg/protocol/consts

File: option.go [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+)
+
+var errLookupNil = errors.New("[Casbin] Lookup is nil")
+
+// Option is the only struct that can be used to set Options.
+type Option struct {
+	F func(o *Options)
+}
+
+type Options struct {
```
Comment (li-jin-gou) [inline option.go:50]: add option unit test

File: option.go [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+func WithUnauthorized(u app.HandlerFunc) Option {
+	return Option{
+		F: func(o *Options) {
+			o.Unauthorized = u
+		},
+	}
+}
+
+func WithForbidden(f app.HandlerFunc) Option {
```
Comment (li-jin-gou) [inline option.go:131]: add comment

**Review by li-jin-gou — CHANGES_REQUESTED** [review]

File: casbin.go (hunk 1 above)
Comment (L2ncE) [reply casbin.go]: done

File: option.go (hunk 2 above)
Comment (L2ncE) [reply option.go:50]: done

File: option.go (hunk 3 above)
Comment (L2ncE) [reply option.go:131]: done

**Review by li-jin-gou — APPROVED** [review]

**Comment by li-jin-gou** [comment]
LGTM

**Pull request merged**